lib/compiler/semantic.js

```diff
@@ -1402,7 +1402,7 @@ class SemanticPass {
         });
 
         if (this.expr_mode === 'assign') {
-            if (!this.scope.is_mutable(node.object.name, this.scope.type)) {
+            if (!node.object.symbol || node.object.symbol.type !== 'var') {
```

Comment:
Would it make sense to introduce a helper with some descriptive name for this condition?